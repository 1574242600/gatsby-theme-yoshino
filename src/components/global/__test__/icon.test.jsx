/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import Icon from "../icon";

describe("Icon 组件", () => {

    it("快照", () => {
        const tree = renderer
            .create(<Icon src={ "/logo/start" } />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("当 加载矢量图成功时", (done) => {
        const wrapper = shallow(<Icon src={ "/logo/succeed" } />);
        setTimeout(() => {
            wrapper.update();
        }, 50);
        

        setTimeout(() => {
            expect(wrapper.html()).toMatch("svg succeed");
            done();
        }, 51);
    });

    it("当 加载矢量图失败时", (done) => {
        const wrapper = shallow(<Icon src={ "/logo/fail" } />);

        setTimeout(() => {
            wrapper.update();
        }, 51);

        setTimeout(() => {
            expect(wrapper.html()).toMatch("load icon error");
            done();
        }, 52);
    });

    beforeAll(() => {
        // eslint-disable-next-line no-undef
        global.fetch = jest.fn();
        global.fetch
            .mockReturnValueOnce(Promise.resolve( { text: () => Promise.resolve("svg succeed")} ))
            .mockReturnValueOnce(Promise.resolve( { text: () => Promise.resolve("svg succeed")} ))
            .mockReturnValueOnce(Promise.resolve( { text: () => Promise.reject() } ));
            
            
    });

    beforeEach(() => {
        fetch.mockClear();
    });
});


