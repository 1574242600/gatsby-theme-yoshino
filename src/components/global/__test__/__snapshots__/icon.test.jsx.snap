// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 组件 快照 1`] = `
<i
  className="icon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
  src="/logo/start"
/>
`;
